File: tests/slotinfo.c
#include <unix.h>
#include <pkcs11.h>
#include <stdio.h>

#include "testlib.h"

TEST_FUNC(slotinfo) {
	CK_SLOT_ID slot;
	CK_SLOT_INFO info;
	int ret;

	check_rv(C_Initialize(NULL_PTR));

	if((ret = find_slot(CK_TRUE, &slot)) != TEST_RV_OK) {
		return ret;
	}

	check_rv(C_GetSlotInfo(slot, &info));

	verify_null(info.slotDescription, 64, 0, "Slot description:\t'%s'\n");
	verify_null(info.manufacturerID, 32, 0, "Manufacturer ID:\t'%s'\n");

	printf("Hardware version: %d.%d\n", info.hardwareVersion.major, info.hardwareVersion.minor);
	printf("Firmware version: %d.%d\n", info.hardwareVersion.major, info.hardwareVersion.minor);
	printf("Token present: %c; Removable: %c; Hardware slot: %c\n", info.flags & CKF_TOKEN_PRESENT ? 'y' : 'n', info.flags & CKF_REMOVABLE_DEVICE ? 'y' : 'n', info.flags & CKF_HW_SLOT ? 'y' : 'n');

	check_rv(C_Finalize(NULL_PTR));

	return TEST_RV_OK;
}

File: doc/sdk/wrappers/C# pkcs11wrappers/pkcs11net/Pkcs11/SlotInfo.cs

using System;
using Net.Sf.Pkcs11.Wrapper;

namespace Net.Sf.Pkcs11
{
	/// <summary>
	/// Description of SlotInfo.
	/// </summary>
	public class SlotInfo
	{
		CK_SLOT_INFO ckSlotInfo;
				
		public string SlotDescription {
			get { return P11Util.ConvertToUtf8String(ckSlotInfo.slotDescription); }
		}
		
		public string ManufacturerID {
			get { return P11Util.ConvertToUtf8String(ckSlotInfo.manufacturerID); }
		}
		
		public Version FirmwareVersion {
			get { return new Version( ckSlotInfo.firmwareVersion); }
		}
		
		public Version HardwareVersion {
			get { return new Version(ckSlotInfo.hardwareVersion); }
		}
		
		public bool IsTokenPresent {
			get { return ((ckSlotInfo.flags & 1L) != 0L); }
		}
		
		
		public bool IsRemovableDevice {
			get { return ((ckSlotInfo.flags & 0x2) != 0L); }
		}
		
		
		public bool IsHwSlot {
			get { return ((ckSlotInfo.flags & 0x4) != 0L); }
		}
		
		
		internal SlotInfo(CK_SLOT_INFO ckSlotInfo) {
			this.ckSlotInfo=ckSlotInfo;
		}
	}
}
